elastic/kibana
# [Discover] Implement ad-hoc data views

## Summary

Closes #137671

This PR enables basic hoc Data Views experience. When action needs persisted data view, the following prompt will be shown. 

**Actions which need persistence of data view currently:**
- Creating an alert
- Share
- Visualise geo fields

![E934D65F-EE40-45EF-886C-A10A7D9B42A9_4_5005_c](https://user-images.githubusercontent.com/39378793/186618136-9d1170e6-6323-4c35-92e3-261d3305680c.jpeg)

Discover Alerts, Share, navigation from lens and adding multiple adhoc data views will be implemented in follow up #139984

### How to test
- Open data view management from discover
- Give a name `k*` and select timestamp field `timefield` to see documents from flights and logs sample data
- Click `Use without saving`

![B6176AC2-FBE4-4F36-B243-653A7A60A03C_1_201_a](https://user-images.githubusercontent.com/39378793/189896243-034ac040-46d0-4fac-8d26-d786614e7807.jpeg)


### Checklist
